common/kafka/kafka.go

```diff
@@ -123,27 +101,93 @@ func getTopic(opts map[string][]string, topicType string) (string, error) {
 	return topic, nil
 }
 
-func getCompression(opts map[string][]string) (proto.Compression, error) {
-	if len(opts[compression]) == 0 {
-		return proto.CompressionNone, nil
+func getCompression(opts url.Values) (kafka.CompressionCodec, error) {
+	if len(opts["compression"]) == 0 {
+		return kafka.CompressionNone, nil
 	}
-	comp := opts[compression][0]
+	comp := opts["compression"][0]
 	switch comp {
 	case "none":
-		return proto.CompressionNone, nil
+		return kafka.CompressionNone, nil
 	case "gzip":
-		return proto.CompressionGzip, nil
+		return kafka.CompressionGZIP, nil
+	case "snappy":
+		return kafka.CompressionSnappy, nil
+	case "lz4":
+		return kafka.CompressionLZ4, nil
 	default:
-		return proto.CompressionNone, fmt.Errorf("Compression '%s' is illegal. Use none or gzip", comp)
+		return -1, fmt.Errorf("Compression '%s' is illegal. Use none, snappy, lz4 or gzip", comp)
```

Comment:
I would hope that it cannot return with a `valid` value, because the error indicates that the rest of the information is void. I suggested without checking the `kafka.CompressionCode` being `int8` to return `nil`. (That would be a perfect value imho.). I think the next good candidate would be `kafka.CompressionUknown`, which does not exists.

Following the call chain, in case of this branch the `kafka` Sink won't be created either way.

To conclude my comment, well I think neither `kafka.CompressionNone` nor `-1` is better then the other, but for different reason, and because there is no better alternative to me either is acceptable in the end.

Reply:
Make sense,  but we should avoid using a number directly.  :-)

Reply:
I personally prefer `kafka.CompressionNone` to `-1`. So i revert to it.